### DISCUSSION TITLE
Missing instrumentation for Apache HttpClient 5?

### DISCUSSION THREAD

**Discussion by elbiocaetano** [open-telemetry/opentelemetry-java-instrumentation#9969] [Q&A]
Hi 👋🏼 ,
We used to instrument Apache v4.3 using the library `io.opentelemetry.instrumentation:opentelemetry-apache-httpclient-4.3`. Now we are upgrading our Apache client to HttpClient v5, but I could not find an instrumentation for this version. I found the java-agent(`io.opentelemetry.javaagent.instrumentation:opentelemetry-javaagent-apache-httpclient-5.0`), but that looks not to have the same classes and features as the instrumentation one.
My question is: can we do the instrumentation as before somehow? Basically we would need a HttpClient with the OpenTelemetry injected in the request, to gather the tracing related to the http requests.
One example on what I want, on instrumentation for apache 4.3:

```java
JerseyClientBuilder jerseyClientBuilder = new JerseyClientBuilder(environment);
    jerseyClientBuilder.setApacheHttpClientBuilder(
        new HttpClientBuilder(environment) {
          @Override
          protected org.apache.http.impl.client.HttpClientBuilder createBuilder() {
            return ApacheHttpClientTelemetry.builder(openTelemetry).build().newHttpClientBuilder();
          }
        });
```

https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-server-auth/src/main/java/org/sdase/commons/server/opa/OpaBundle.java#L182

Thanks in advance.

**Comment by trask** [comment]
hi @elbiocaetano! you're right, no one has implemented library instrumentation for Apache HttpClient 5 yet, but it's definitely something we would accept as a contribution if you or someone else wants to implement it

**Reply by elbiocaetano** [reply]
Thanks for the kind answer, I have created an [issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/10099) and a [PR](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10100) with the instrumentation for apache httpclient 5. I would appreciate if you could take a look. Thank you very much.
